python/pyspark/ml/tests.py

```diff
@@ -455,6 +455,14 @@ def test_logistic_regression_check_thresholds(self):
             LogisticRegression, threshold=0.42, thresholds=[0.5, 0.5]
         )
 
+    def test_preserve_set_state(self):
+        model = Binarizer()
+        self.assertFalse(model.isSet("threshold"))
+        model._transfer_params_to_java()
```

Comment:
Would it make sense to do an actual transform here instead of the two inner parts of the transform?

Reply:
I think that would be a reasonable thing to do, the slight increase in testing overhead is probably worth it, it keeps us from being too closely tied to the implementation details and we already use `SparkSessionTestCase` in a lot of places.